services/provider/server/server.go

```diff
@@ -367,6 +370,35 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 		}),
 	})
 
+	if consumerResource.Spec.StorageQuotaInGiB > 0 {
+		clusterResourceQuotaSpec := &quotav1.ClusterResourceQuotaSpec{
+			Selector: quotav1.ClusterResourceQuotaSelector{
+				LabelSelector: &metav1.LabelSelector{
+					MatchExpressions: []metav1.LabelSelectorRequirement{
+						{
+							Key:      "dummy",
+							Values:   []string{"dummy"},
+							Operator: metav1.LabelSelectorOpNotIn,
+						},
+					},
+				},
+			},
```

Comment:
Why do we need a selector here and why is the value called "dummy"?

Reply:
ClusterResourceQuotaSelector is used to select projects.  At least one of LabelSelector or AnnotationSelector is mandatory. There is no option to select all projects. this is the one way to mitigate as of now.

I have discussed this with Leela as well and we 3 options as of now.
1. Same as above "guess" some `key-vaule` may not exist and apply the selector
2. Edit Project template if there is any in the cluster to automatically add an `annotation/label` when new project is created [project.config.openshift.io/cluster](https://red-hat-storage.github.io/ocs-training/training/ocs4/ocs4-cluster-storage-quotas.html#_creating_a_project_template_that_includes_storage_annotation)
3. Extend the webhook in the client operator to trap namespace creation and add annotation/label